package/etc/conf.d/conflib/syslog/app-syslog-barracuda_waf.conf

```diff
@@ -3,16 +3,54 @@ block parser app-syslog-barracuda_waf() {
         rewrite {
             r_set_splunk_dest_default(
                 index("netwaf")
-                sourcetype('barracuda:waf')
                 vendor("barracuda")
                 product("syslog")
+                sourcetype('barracuda:syslog')
             )
         };
+
+        if (program('SYS')) {
+            rewrite {
+                r_set_splunk_dest_update_v2(
+                    sourcetype('barracuda:system')
+                );
+            };
+        } elif (program('WF')) {
+            rewrite {
+                r_set_splunk_dest_update_v2(
+                    sourcetype('barracuda:waf')
+                );
+            };   
+        } elif (program('TR')) {
+            rewrite {
+                r_set_splunk_dest_update_v2(
+                    sourcetype('barracuda:web')
+                );
+            };   
+        } elif (program('AUDIT')) {
+            rewrite {
+                r_set_splunk_dest_update_v2(
+                    sourcetype('barracuda:audit')
+                );
+            };   
+        } elif (program('NF')) {
+            rewrite {
+                r_set_splunk_dest_update_v2(
+                    sourcetype('barracuda:firewall')
+                );
+            };   
+        };
     };
 };
+
+
 application app-syslog-barracuda_waf[sc4s-syslog] {
```

Comment:
if we are filtering based on program , the preferred topic is [sc4s-syslog-pgm]

Reply:
done